examples/write_and_read_streams.cpp

```diff
@@ -0,0 +1,289 @@
+    auto string_array = sp::string_array(std::move(string_values));
+
+    // Create record batch with named columns (same schema for all batches)
+    return sp::record_batch(
+        {{"id", sp::array(std::move(int_array))},
+         {"value", sp::array(std::move(float_array))},
+         {"flag", sp::array(std::move(bool_array))},
+         {"name", sp::array(std::move(string_array))}}
+    );
+}
+
+/**
+ * Verify that two sets of record batches are identical
+ * Returns true if all batches match, false otherwise
+ */
+bool verify_batches_match(
+    const std::vector<sp::record_batch>& original_batches,
+    const std::vector<sp::record_batch>& deserialized_batches
+)
+{
+    if (original_batches.size() != deserialized_batches.size())
+    {
+        std::cerr << "ERROR: Batch count mismatch! Original: " << original_batches.size()
+                  << ", Deserialized: " << deserialized_batches.size() << "\n";
+        return false;
+    }
+
+    bool all_match = true;
+    for (size_t batch_idx = 0; batch_idx < original_batches.size(); ++batch_idx)
+    {
+        const auto& original = original_batches[batch_idx];
+        const auto& deserialized = deserialized_batches[batch_idx];
+
+        // Check basic structure
+        if (original.nb_columns() != deserialized.nb_columns() || original.nb_rows() != deserialized.nb_rows())
+        {
+            std::cerr << "ERROR: Batch " << batch_idx << " structure mismatch!\n";
+            all_match = false;
+            continue;
+        }
+
+        // Check column names
+        if (!std::ranges::equal(original.names(), deserialized.names()))
+        {
+            std::cerr << "WARNING: Batch " << batch_idx << " column names mismatch!\n";
+        }
+
+        // Check column data
+        for (size_t col_idx = 0; col_idx < original.nb_columns(); ++col_idx)
+        {
+            const auto& orig_col = original.get_column(col_idx);
+            const auto& deser_col = deserialized.get_column(col_idx);
+
+            if (orig_col.data_type() != deser_col.data_type())
+            {
+                std::cerr << "ERROR: Batch " << batch_idx << ", column " << col_idx << " type mismatch!\n";
+                all_match = false;
+                continue;
+            }
+
+            // Check values
+            for (size_t row_idx = 0; row_idx < orig_col.size(); ++row_idx)
+            {
+                if (orig_col[row_idx] != deser_col[row_idx])
+                {
+                    std::cerr << "ERROR: Batch " << batch_idx << ", column " << col_idx << ", row " << row_idx
+                              << " value mismatch!\n";
+                    std::cerr << "  Original: " << orig_col[row_idx]
+                              << ", Deserialized: " << deser_col[row_idx] << "\n";
+                    all_match = false;
+                }
+            }
+        }
+    }
+
+    return all_match;
+}
+
+int main()
+{
+    std::cout << "=== Sparrow IPC Stream Write and Read Example ===\n";
```

Comment:
Maybe I missed it but it seems we're missing reading from an already existing stream file? (from the integration tests for example?) and also maybe write into a stream file?

Reply:
Fixed